guillotina/db/storages/pg.py

```diff
@@ -546,3 +553,18 @@ def _db_transaction_factory(self, txn):
             smt = await txn._db_conn.prepare(NUM_RESOURCES)
             result = await smt.fetchval()
         return result
+
+    async def get_total_resources_of_type(self, txn, type_):
+        async with txn._lock:
+            smt = await txn._db_conn.prepare(NUM_RESOURCES_BY_TYPE)
+            result = await smt.fetchval(type_)
+        return result
+
+    # Massive treatment without security
+    async def _get_resources_of_type(self, txn, type_):
```

Comment:
I would recommend using something like the iterate_keys method. Trust me, on large data sets, a cursor will die on you.